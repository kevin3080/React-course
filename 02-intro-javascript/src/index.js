//operador condicional tarnario:

const activo = true;

// let mensaje = '';

// if (!activo){ // si le coloco ! es la negacion de ese atributo
//     mensaje= 'Activo';
// }else{
//     mensaje = 'Inactivo';
// }


// Es igual a:
// const mensaje = (!activo) ? 'Activo' : 'Inactivo';
//const mensaje = (!activo) ? 'Activo' : null;
//otra forma corta de hacer un if:
const mensaje = activo && 'Activo'
const mensaje1 = !activo && 'Activo'
console.log(mensaje);
console.log(mensaje1);