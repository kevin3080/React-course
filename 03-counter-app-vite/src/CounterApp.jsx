import { useState } from 'react'; // hooks
import PropTypes from 'prop-types';


export const CounterApp = ({ value }) => {

    // const handleAdd = (event, newValue) => {
    //    // console.log(event)
    //    console.log(newValue);
    // };

    // Esto es un hook:
    const [ counter, setCounter ] = useState( value )
    // cada vez que se actualiza el estado se vuelve a lanzar el componente
    const handleAdd = () => {
        // console.log('1')

        //Llamamos al hook para cambiar el valor del estado
        // setCounter(counter + 1);
        //otra manera:
        setCounter( (c) => c + 1 );
        
     };

     const decrementar = () => setCounter( counter - 1 );
    

     const reset = () => setCounter ( (resetear) => value );

  return (
    <>
        <h1>CounterApp</h1>
        <h2>{ counter }</h2>

        <button onClick={handleAdd}> +1 </button>
        <button onClick={decrementar}> -1 </button>
        <button aria-label='btn-reset' onClick={reset}> Reset </button>
    
    </>
  )
};
// <button onClick={ (event)=> handleAdd(event, 'hola') }>
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
  }
  //dato curioso: Primero se ejecuto los PropsApp.defaultProps antes de los PropsApp.propTypes
  CounterApp.defaultProps = {
    value: 'No hay numero'
  }
  



