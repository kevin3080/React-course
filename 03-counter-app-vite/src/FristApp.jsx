
//shorcout: rafc

//import { Fragment } from "react"

const sda = (a,b) =>  a*b;

const newMessage = [1,2,3,4,5,6,7,8,9];
const newMessage2 = {
    message: 'Hola Mundo',
    title: 'Pernia'
};
export const FristApp = () => {
  return (
    <>
        <h2>{newMessage}</h2>
        {/* <code>{JSON.stringify(newMessage2)}</code> */}
        <h1>Kevin</h1>
        <p>Soy un subtitulo</p>
        <h2>{sda(43,655)}</h2>
    </>
    
    
  )
}
// JSON.stringify(aca puedes mostrar codigo javascript)
//si no quieres importar el fragmento solo agregar <></>
