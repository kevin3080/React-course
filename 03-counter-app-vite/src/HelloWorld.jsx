// export function App() {
//    // document.createElement...
//     return <h1>Hola Mundo!!!!</h1>
// }
// export default App;
//Los funcional components van capitalizados con PascalCase o OpenCamelCase

import React from 'react'
// funcional components:
export const HelloWorldApp = () => {
  return (
    <h1>Hello World App!!!</h1>
  )
}
