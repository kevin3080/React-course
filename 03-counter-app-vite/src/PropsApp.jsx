import PropTypes from 'prop-types';
// En yarn hay que intalar este paquete en consola: yarn add prop-types 


export const PropsApp = ({title, subTitle, name}) => {
 
    return (
    <>
        <h1 data-testid="test-title">{title}</h1>
        <h2>{name}</h2>
        <h2>{subTitle}</h2>
    </>
    
  )
}

PropsApp.propTypes = {
  title: PropTypes.string.isRequired,
  //Agregandole isRequired esta obligando a que se mande un titulo
  subTitle: PropTypes.string.isRequired
}
//dato curioso: Primero se ejecuto los PropsApp.defaultProps antes de los PropsApp.propTypes
PropsApp.defaultProps = {
  name: 'Kevin Pernia',
  subTitle: 'No hay subtitulo',
  title: 'No hay titulo',
}
