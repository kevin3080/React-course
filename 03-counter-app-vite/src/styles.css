hmtl, body{
    background-color: #21232A;
    color: white;
    font-family: Helvatica, Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button{
    padding: 10px;
    border-radius: 10px;
    font-size: 1.3rem;
    margin-right: 5px;
}