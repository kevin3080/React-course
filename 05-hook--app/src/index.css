body{
    padding: 20px;
}

button{
    margin-right: 10px;
}

.input-group{
    width: 30% !important;
}