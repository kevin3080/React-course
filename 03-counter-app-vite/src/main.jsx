import React from 'react';
import ReactDOM from "react-dom/client";
import { CounterApp } from './CounterApp';
 import { FristApp } from './FristApp';
import { PropsApp } from './PropsApp';
// import { HelloWorldApp } from './HelloWorld';
// import { PropsApp } from './PropsApp';
//import {App} from './FristApp'
import './styles.css';




// renderizamos:
// Se aconseja que siempre trabajemos en el modo estricto
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        
      <CounterApp value = {100} />
      {/* <PropsApp title='Frist App' subTitle='subtitulo'/> */}
    </React.StrictMode>
)
// todo lo que se coloca como atributo en los componentes son props
