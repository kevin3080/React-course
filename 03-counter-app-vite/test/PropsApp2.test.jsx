import { getByText, render, screen } from "@testing-library/react"
import { PropsApp } from "../src/PropsApp"


describe('Pruebas en PropsApp', () => { 

    const title = 'Frist App';
    const subTitle = 's';

    test('debe de hacer match con el snapshot', () => { 


        const { container } = render( <PropsApp title={ title } subTitle={subTitle}/> )
        expect ( container ).toMatchSnapshot();
    });

     test('Debe de mostrar el mensaje Frist App', () => { 

         render(<PropsApp title={title} subTitle={subTitle}/>)
        // screen.debug(); // con esto puedes ver lo que hay en el <body></body>
        expect(screen.getByText(title)).toBeTruthy(); // que exista el title
        //expect(screen.getByText(title)).not.toBeTruthy(); // que no exista el title

    });

      test('debe de mostar el titulo en un h1', () => { 
        render(<PropsApp title={title} subTitle={subTitle}/>)
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);

    });

    test('debe de mostrar el subitulo enviado por props', () => { 
       render(<PropsApp 
       title={title}
       subTitle={subTitle}
       /> );
     
     })
 });