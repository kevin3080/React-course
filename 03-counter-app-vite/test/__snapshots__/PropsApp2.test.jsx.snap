// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PropsApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Frist App
  </h1>
  <h2>
    Kevin Pernia
  </h2>
  <h2>
    s
  </h2>
</div>
`;
